src/progressive-profiling/ProgressiveProfiling.jsx

```diff
@@ -57,27 +55,22 @@ const ProgressiveProfiling = (props) => {
   const registrationEmbedded = isHostAvailableInQueryParams();
 
   const queryParams = getAllPossibleQueryParams();
-  const authenticatedUser = getAuthenticatedUser();
-  const DASHBOARD_URL = getConfig().LMS_BASE_URL.concat(DEFAULT_REDIRECT_URL);
+  const authenticatedUser = getAuthenticatedUser() || location.state?.authenticatedUser;
   const enablePopularAndTrendingRecommendations = getConfig().ENABLE_POPULAR_AND_TRENDING_RECOMMENDATIONS;
 
   const [registrationResult, setRegistrationResult] = useState({ redirectUrl: '' });
   const [formFieldData, setFormFieldData] = useState({ fields: {}, extendedProfile: [] });
-  const [canViewWelcomePage, setCanViewWelcomePage] = useState(false);
   const [values, setValues] = useState({});
   const [showModal, setShowModal] = useState(false);
   const [showRecommendationsPage, setShowRecommendationsPage] = useState(false);
 
   useEffect(() => {
-    configureAuth(AxiosJwtAuthService, { loggingService: getLoggingService(), config: getConfig() });
-    ensureAuthenticatedUser(DASHBOARD_URL)
```

Comment:
What of the purpose of ensureAuthenticatedUser was to redirect to provided URL incase the user is not authenticated?
https://github.com/openedx/frontend-platform/blob/master/src/auth/AxiosJwtAuthService.js#L250

How are we handling it now? Is it being handled by the condition below?
```
if (
    !(location.state?.registrationResult || registrationEmbedded)
    || welcomePageContextApiStatus === FAILURE_STATE
    || (welcomePageContextApiStatus === COMPLETE_STATE && !Object.keys(welcomePageContext).includes('fields'))
  ) {
    const DASHBOARD_URL = getConfig().LMS_BASE_URL.concat(DEFAULT_REDIRECT_URL);
    global.location.assign(DASHBOARD_URL);
    return null;
  }
```

If it is getting handled, i would still have a feeling that we should have a check if user is not authenticated, then return to Dashboard.

Reply:
I will confirm this.

Reply:
I checked and the user was not being redirected to login page when they were unauthenticated instead they were able to see an empty welcome page.
This has now been fixed in the latest commit.